sound/soc/sof/ipc4-topology.c

```diff
@@ -1419,13 +1419,55 @@ static int sof_ipc4_control_load_volume(struct snd_sof_dev *sdev, struct snd_sof
 	return 0;
 }
 
+static int sof_ipc4_control_load_bytes(struct snd_sof_dev *sdev, struct snd_sof_control *scontrol)
+{
+	struct sof_ipc4_control_data *control_data;
+	int ret;
+
+	scontrol->size = sizeof(struct sof_ipc4_control_data) + scontrol->priv_size;
+
+	scontrol->ipc_control_data = kzalloc(scontrol->size, GFP_KERNEL);
+	if (!scontrol->ipc_control_data)
+		return -ENOMEM;
+
+	control_data = scontrol->ipc_control_data;
+	control_data->index = scontrol->index;
+	if (scontrol->priv_size > 0) {
+		memcpy(control_data->data, scontrol->priv, scontrol->priv_size);
+
+		if (control_data->data->magic != SOF_IPC4_ABI_MAGIC) {
+			dev_err(sdev->dev, "Wrong ABI magic 0x%08x.\n", control_data->data->magic);
+			ret = -EINVAL;
+			goto err;
+		}
```

Comment:
Currently we do not have `SND_SOC_TPLG_CTL_BYTES` in tplg2, right? They can be added after this PR and then they will have the proper header and magic?

Reply:
We don't have ```SND_SOC_TPLG_CTL_BYTES``` in tplg2 now. The proper header and magic are generated by sof-ctl, which is the current solution for tplg1. I have a patch for sof-ctl to support the new header. After these kernel patches are merged, I will submit the patch to sof-ctl correspondingly.